src/charm.py

```diff
@@ -365,6 +396,72 @@ def open_mongos_port(self) -> None:
         """Opens the mongos port for TCP connections."""
         self.unit.open_port("tcp", Config.MONGOS_PORT)
 
+    def is_role(self, role_name: str) -> bool:
```

Comment:
hmm, I can do that, but I would like to keep parity with Charmed MongoDB which uses `is_role`. This function is one of the major functions of the sharding implementation in mongos and mongodb charms

If we change this to `is_of_role` then we must change:
1. all lib code of mongos + mongodb charm as they use this function
2. the role related functions in mongos + mongodb charm 
3. relevant unit tests

Given this scope it would have to be outside of this PR. If you would like this , we can make it happen, but as a separate PR. if so, would you mind making a Jira ticket?